src/ert/ensemble_evaluator/dispatch.py

```diff
@@ -1,107 +1,106 @@
 import asyncio
-import contextlib
 import logging
+import threading
 import time
 import traceback
 from collections import OrderedDict, defaultdict
+from typing import Callable, Mapping
 
 from ert.ensemble_evaluator import identifiers
 
 logger = logging.getLogger(__name__)
 
 
-class BatchingDispatcher:
-    def __init__(self, loop, timeout, max_batch=1000):
-        self._timeout = timeout
+class BatchingDispatcher:  # pylint: disable=too-many-instance-attributes
+    def __init__(self, sleep_between_batches_seconds, max_batch=1000):
+        self._sleep_between_batches_seconds = sleep_between_batches_seconds
         self._max_batch = max_batch
 
-        self._LOOKUP_MAP = defaultdict(list)
+        self._LOOKUP_MAP: Mapping[set, Callable] = defaultdict(lambda: (lambda _: None))
```

Comment:
As discussed, it would be nice to have a default function, which says "event not handled" instead of just an empty function. But I like the simplification :)

Reply:
Should probably be `Mapping[str, Callable]` also ert.ensemble_evaluator.identifiers should not be a package but a [`StrEnum`](https://docs.python.org/3/library/enum.html#enum.StrEnum) (see for instance `ConfigKeys` for how to set that up with python 3.8).

Reply:
> Also, this used to default to a list so that you could have multiple even handlers for each event which is a common assumption. So probably change this to Mapping[str, List[Callable[[], Any]]

This is sort-of the point of simplification that currently we do have only a single event handler function. Basically if we need to have multiple handler functions than we can adjust I guess.

Reply:
> > Also, this used to default to a list so that you could have multiple even handlers for each event which is a common assumption. So probably change this to Mapping[str, List[Callable[[], Any]]
> 
> This is sort-of the point of simplification that currently we do have only a single event handler function. Basically if we need to have multiple handler functions then we can adjust I guess.

Then you probably want to rename `add` to `set`

Reply:
That sounds like a good suggestion @valentin-krasontovitsch

Reply:
renamed method, default handler changed to logger